### PR TITLE
Basic HTTP Authentication via htpasswd

### PR DESCRIPTION
**Pull request by hey-johnnypark**
Hi, 

I think this option might me useful when deploying to cloud instances such as Heroku. Please merge if you think this options fits into live-server. I will use locally only otherwise. 

Cheers
Andi

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ LiveServer.start = function(options) {
 	var staticServerHandler = staticServer(root);
 	var wait = options.wait || 0;
 	var browser = options.browser || null;
+	var authPath = options.authPath || null;
```
Comment (tapio) [inline index.js]: I think we should use htpasswd (or authPath, but I think I prefer the former) for both the cli flag and the api param for consistency reasons (I'm aware that not all current params actually are consistent :disappointed:).

File: package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
 		"object-assign": "latest",
 		"opn": "latest",
 		"send": "latest",
-		"watchr": "2.3.x"
+		"watchr": "2.3.x",
+		"http-auth": "2.2.8"
```
Comment (tapio) [inline package.json]: Probably a good idea to allow patch updates by using 2.2.x.

File: test/cli.js [hunk 3]
```diff
@@ -45,4 +45,15 @@ describe('command line usage', function() {
...
+		exec(cmd,
+			[ "--htpasswd=" + path.join(__dirname, "data/htpasswd-test"),
+				"--no-browser",
+				"--test"
+			], opts, function(error, stdout, stdin) {
+			assert(!error, error);
+			assert(stdout.indexOf("Serving") == 0, "serving string not found");
+			done();
+		});
+	});
```
Comment (tapio) [inline test/cli.js]: I think this test is not particularly useful test case as it only tests that giving the param does not throw some error. There should probably be a dedicated "htpasswd.js" test file, similar to "mount.js" that creates the server and then tests that it returns the proper HTTP 401 status code and WWW-Authenticate header, and if possible also tests right and wrong passwords.

**Comment by tapio** [comment]
Thanks for the contribution and sorry for taking this long to review. I added some inline comments to the diff - please take a look and tell me what you think and if/if not you have time to act upon them.

**Comment by hey-johnnypark** [comment]
Hi,

I implemented the optimisations you suggested, please check. 

Cheers
Johnny

**Comment by tapio** [comment]
Thanks, merged (with some minor tweaks)!

**Pull request merged**